### PR TITLE
refactor(rxSpinner): components/rxSpinner to elements/Spinners

### PR DESCRIPTION
**Pull request by khaziwallis**
JIRA: https://jira.rax.io/browse/EO-1091

### LGTMs
- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM
- [x] QE LGTM

Before
---------
Components > rxSpinner
---------
screen shot 2016-07-21 at 7 00 17 pm
screen shot 2016-07-21 at 6 58 57 pm

After
---------
Elements > Spinners
---------
screen shot 2016-07-21 at 7 02 25 pm

Full Page
---------
Elements > Spinners
---------
screen shot 2016-07-21 at 7 04 05 pm

When view All Modules
---------
 And filter by "spinner"
---------
screen shot 2016-07-21 at 7 07 07 pm

 And filter by "button"
---------
screen shot 2016-07-21 at 7 08 11 pm

 And filter by "label"
---------
screen shot 2016-07-21 at 7 09 56 pm

 And filter by "progress"
---------
screen shot 2016-07-21 at 7 10 17 pm

 And filter by "pending"
---------
screen shot 2016-07-21 at 7 10 50 pm

 And filter by "loading"
---------
screen shot 2016-07-21 at 7 11 02 pm

 And filter by "rxSpinner"
---------
screen shot 2016-07-21 at 7 11 14 pm

API DOCS
------------
screen shot 2016-07-21 at 9 39 33 pm

@Droogans As per the suggestion , updated the changes and also attached the screen short please review

### CODE REVIEW

**Comment by manojdas2212** [comment]
Before
---------
Components > rxSpinner
---------
screen shot 2016-07-21 at 7 00 17 pm
screen shot 2016-07-21 at 6 58 57 pm

After
---------
Elements > Spinners
---------
screen shot 2016-07-21 at 7 02 25 pm

Full Page
---------
Elements > Spinners
---------
screen shot 2016-07-21 at 7 04 05 pm

When view All Modules
---------
 And filter by "spinner"
---------
screen shot 2016-07-21 at 7 07 07 pm

 And filter by "button"
---------
screen shot 2016-07-21 at 7 08 11 pm

 And filter by "label"
---------
screen shot 2016-07-21 at 7 09 56 pm


 And filter by "progress"
---------
screen shot 2016-07-21 at 7 10 17 pm

 And filter by "pending"
---------
screen shot 2016-07-21 at 7 10 50 pm

 And filter by "loading"
---------
screen shot 2016-07-21 at 7 11 02 pm

 And filter by "rxSpinner"
---------
screen shot 2016-07-21 at 7 11 14 pm

@Droogans As per the suggestion , updated the changes and also attached the screen short please review

File: src/elements/Spinner/Spinner.meta.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "description": "Adds a spinner icon to any element",
+    "isLegacy": true,
+    "keywords": [
+    	"spinner",
+    	"button",
+    	"label",
+    	"progress",
+    	"pending",
+    	"loading",
+    	"rxSpinner"
```
Comment (100stacks) [inline src/elements/Spinner/Spinner.meta.json]: The keywords should be listed in alphabetical order.

File: src/elements/Spinner/docs/Spinner.docs.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<p>
+  A component adds a spinner icon to any element
+</p>
+
+<rx-example name="spinner"></rx-example>
```
Comment (100stacks) [inline src/elements/Spinner/docs/Spinner.docs.html]: @khaziwallis @manojdas2212: Please update to `name="spinner.simple"`.

File: src/elements/Spinner/docs/examples/spinner.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div ng-controller="rxSpinnerCtrl">
+<div ng-controller="SpinnerCtrl">
```
Comment (100stacks) [inline src/elements/Spinner/docs/examples/spinner.html]: @khaziwallis @manojdas2212: Please update to `SpinnerSimpleCtrl` and change the file name to `spinner.simple.html`.

File: src/elements/Spinner/docs/examples/spinner.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
 angular.module('demoApp')
-.controller('rxSpinnerCtrl', function ($scope) {
+.controller('SpinnerCtrl', function ($scope) {
```
Comment (100stacks) [inline src/elements/Spinner/docs/examples/spinner.js]: @khaziwallis @manojdas2212: Please update to `SpinnerSimpleCtrl` and change the file name to `spinner.simple.js`.

File: src/utilities/urlUtils/scripts/urlUtils.spec.js [hunk 5]
```diff
@@ -192,18 +193,18 @@ describe('utilities:urlUtils', function () {
             var parsed = urlutils.parseUrl(undefined);
             expect(parsed.protocol).to.equal('http:');
             expect(parsed.hostname).to.equal('localhost');
-            expect(parsed.port).to.equal('9877');
+            expect(parsed.port).to.equal('9878');
```
Comment (100stacks) [inline src/utilities/urlUtils/scripts/urlUtils.spec.js]: @khaziwallis @manojdas2212:  What was the reason for the port change?

File: src/elements/Spinner/Spinner.meta.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
+    "displayName": "Spinners",
+    "stability": "stable",
+    "description": "Adds a spinner icon to any element",
+    "isLegacy": true,
```
Comment (100stacks) [inline src/elements/Spinner/Spinner.meta.json]: Change from:
```js
 "isLegacy": true,
```
to:
```js
"api": "directive:rxSpinner",
```

This change is what tells the build process to stop using the deprecated demo styles.

**Comment by manojdas2212** [comment]
API DOCS
------------
screen shot 2016-07-21 at 9 39 33 pm

**Comment by manojdas2212** [comment]
@100stacks updated changes please take a look

**Comment by glynnis** [comment]
This LGTM from a design standpoint! Thanks @khaziwallis

**Comment by 100stacks** [comment]
You will need to squash and rebase your commits.

**Comment by 100stacks** [comment]
Also, please remember to squash this PR to two commits:
* first commit - just move files
* second commit - contains all edits/changes

**Comment by manojdas2212** [comment]
@Droogans @100stacks @khaziwallis Can you please review my PR and also not able to do squash and rebase as i dont have access

File: src/elements/Spinner/Spinner.meta.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    "description": "Adds a spinner icon to any element",
+    "api": "directive:rxSpinner",
+    "keywords": [
+    	"button",
+    	"label",
+        "loading",
+    	"progress",
+    	"pending",
+    	"rxSpinner",
+        "spinner"
```
Comment (100stacks) [inline src/elements/Spinner/Spinner.meta.json]: @manojdas2212, minor update...please fix indentation for `keywords`.  Namely for `loading` and `spinner`.

**Comment by manojdas2212** [comment]
@100stacks Please take a look for review , Dev LGTM required on this

**Comment by CITguy** [comment]
LGTM

**Comment by CITguy** [comment]
@khaziwallis: please squash to 1 commit

**Comment by manojdas2212** [comment]
@CITguy squash done please take a look

**Comment by CITguy** [comment]
Thanks @manojdas2212!

**Pull request merged**